docs/api.rst

```diff
@@ -0,0 +1,122 @@
+Vumi HTTP Retry API
+===================
+
+Contents
+--------
+
+- :ref:`response-format-overview`
+- :ref:`api-methods`
+
+    - :http:post:`/requests/`
+
+.. _response-format-overview:
+
+Response Format Overview
+------------------------
+
+Successful responses to GET requests will contain the requested data in json
+format.
+
+**Example response (success response)**:
+
+.. sourcecode:: http
+
+    HTTP/1.1 200 OK
+    {
+        ...
+    }
+
+Errors are returned with the relevant HTTP error code and a json object
+containing an array of the relevant ``errors``. Each error object in the array
+has a ``type`` string to identify the error type, and a ``message`` string to be
+used as a human-readable message for the error.
+
+**Example error response**:
+
+.. sourcecode:: http
+
+    HTTP/1.1 400 Bad Request
+    {
+        "errors": [{
+          "type": "not_enough_coffee",
+          "message": "Not enough coffee"
+        }]
+    }
+
+
+.. _api-methods:
+
+API Methods
+-----------
+
+.. http:post:: /requests/
+
+   Adds a new request to retry.
+
+   :reqheader Accept: Should be ``application/json``.
+   :reqheader X-Owner-ID: The id of the owner to associate with this request. A
+   maximum of ``10000`` request retries are allowed per owner per 30 minutes.
+
+   :jsonparam array intervals:
+       The second-based intervals at which retries should be done. Defaults to
+       ``[30, 300, 900]``.
+
+   :jsonparam object request:
+       The request to retry.
+
+   :jsonparam str request.url:
+       The url to make the request to.
+
+   :jsonparam str request.method:
+       The request's http method.
+
+   :jsonparam object request.headers:
+       Optional object of headers to add to the request.
+
+   :jsonparam str request.body:
+       Optional request body.
+
+   :resheader Content-Type: ``application/json``.
```

Comment:
Missing a description of [error code 429](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#429) here, sphinx won't build since it doesn't recognise the code. I've added an example of the error at the bottom though. Also happy to mention in some other way.